archethic-foundation/archethic-node
# Use CAStore to check the HTTPS certificate in Coingecko calls

# Description

Ensure the verification of SSL certificates using :httpc client when OracleChain fetches Coingecko UCO's price.

Fixes #591

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

- Edit `/etc/hosts` by adding a new line:
`127.0.0.1 localhost api.coingecko.com`

- Navigate to `/tmp/ `
- Create a new certificate: `openssl req -x509 -nodes -days 365 -sha256 -newkey rsa:2048 -keyout mykey.pem -out mycert.pem`
- Start an HTTP server with the certificate: `openssl s_server -accept 443 -cert mycert.pem -key mykey.pem -www &`
- Start a node 
- You should see this error during the OracleChain polling
<img width="1274" alt="image" src="https://user-images.githubusercontent.com/14906964/192990451-b05af9bc-7854-47a9-82d3-05b5ad46ab47.png">

By reverting the steps, you should not see any warnings regarding OracleChain polling, as the SSL certificates verify the validity of Coingecko's SSL certificate. 

# Checklist:

- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules
